src/iocore/net/BIO_fastopen.cc

```diff
@@ -167,18 +249,18 @@ fastopen_bread(BIO *bio, char *out, int outsz)
 static long
 fastopen_ctrl(BIO *bio, int cmd, long larg, void *ptr)
 {
+#if !defined(OPENSSL_IS_BORINGSSL) && (OPENSSL_VERSION_MAJOR >= 3)
```

Comment:
It'd be nice if we could avoid using `OPENSSL_IS_BORINGSSL`. Otherwise I'd have to add `OPENSSL_IS_AWSLC`. What's the context of this check?

Reply:
The OPENSSL_IS_BORINGSSL is already used in many places in ATS code.  It is controlled by this cmake code:  https://github.com/apache/trafficserver/blob/master/cmake/CheckOpenSSLIsBoringSSL.cmake .

Reply:
That doesn't answer my question. I know the check exists and you are not the only person who use it. I had to add one for AWS-LC because `OPENSSL_IS_BORINGSSL` is used. If everything is done by checks for technical requirements (i.e. not by the library name), I didn't have to do that.

My question is why this block is only needed by OpenSSL 3+.

Reply:
It's necessary to avoid triggering the ink_assert(), see below.

Reply:
I wan't aware we also have to support a third TLS library.

Reply:
As the community we don't have the consensus to support AWS-LC and other variations. It's just I don't think checking the library name is not a great idea in general where there are many forks of SSL library. That's why I was asking just the context so that I can find a better way later.